### PR TITLE
New component: text LCD

### PR DESCRIPTION
**Pull request by RREE**
This patch add simple text based LCDs based on the HD44780 driver. Almost all text LCD use this or a compatible driver module. 

This first version uses a PCF8574 I/O expander via I²C. I can also add a 4bit implementation, but that requires 7 or 8 GPIOs of the controller. That's why I don't use it anymore.

As a prerequisite this patch requires the merge of the aforementioned PCF8574 pull request.

### CODE REVIEW

**Comment by RREE** [comment]
I created the branch in my copy of the ADL off the branch for the pcf8574.  It seems that this pull request contains both the pcf8574 and the new LCD code

File: components/src/io_expander/pcf8574/pcf8574.ads [hunk 1]
```diff
@@ -0,0 +1,38 @@
+--
```
Comment (Fabien-Chouteau) [inline components/src/io_expander/pcf8574/pcf8574.ads:1]: Same here.

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]
Hi @RREE, thanks for the contribution.
Besides the header this looks good to me.

I guess I can close the other PR for the PCF8574.

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]
There's an unreferenced with:
```
lcd_hd44780-pcf8574.ads:35:09: warning: unit "HAL.I2C" is not referenced 
```

You should run `scripts/build_all_examples.py` to check your patch.

**Review by Fabien-Chouteau — APPROVED** [review]

**Comment by Fabien-Chouteau** [comment]
Thank you @RREE !

**Pull request merged**